From mcarlson@broadcom.com  Thu Jun 23 22:01:21 2011
From: Matt Carlson <mcarlson@broadcom.com>
Date: Wed, 22 Jun 2011 18:56:52 -0700
Subject: [PATCH 011/194] tg3: Push phylib definitions to phylib
Git-commit: 6a443a0f72ad7706345412dbd2e4d4981fdfce39
Patch-mainline: v2.6.34-rc1
References: bnc#697783, FATE#311457

This patch pushes phylib definitions out to phylib headers.  For phy
IDs, this removes some code duplication.

Signed-off-by: Matt Carlson <mcarlson@broadcom.com>
Reviewed-by: Michael Chan <mchan@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 drivers/net/tg3.c        |   50 +++++++++++++++++++++++-----------------------
 drivers/net/tg3.h        |   15 +++++--------
 drivers/net/tg3_compat.h |   32 +++++++++++++++++++++++++++++
 3 files changed, 63 insertions(+), 34 deletions(-)

diff --git a/drivers/net/tg3.c b/drivers/net/tg3.c
index 7179ee2..0b1dfab 100644
--- a/drivers/net/tg3.c
+++ b/drivers/net/tg3.c
@@ -975,17 +975,17 @@ static void tg3_mdio_config_5785(struct tg3 *tp)
 
 	phydev = tp->mdio_bus->phy_map[TG3_PHY_MII_ADDR];
 	switch (phydev->drv->phy_id & phydev->drv->phy_id_mask) {
-	case TG3_PHY_ID_BCM50610:
-	case TG3_PHY_ID_BCM50610M:
+	case PHY_ID_BCM50610:
+	case PHY_ID_BCM50610M:
 		val = MAC_PHYCFG2_50610_LED_MODES;
 		break;
-	case TG3_PHY_ID_BCMAC131:
+	case PHY_ID_BCMAC131:
 		val = MAC_PHYCFG2_AC131_LED_MODES;
 		break;
-	case TG3_PHY_ID_RTL8211C:
+	case PHY_ID_RTL8211C:
 		val = MAC_PHYCFG2_RTL8211C_LED_MODES;
 		break;
-	case TG3_PHY_ID_RTL8201E:
+	case PHY_ID_RTL8201E:
 		val = MAC_PHYCFG2_RTL8201E_LED_MODES;
 		break;
 	default:
@@ -1135,12 +1135,12 @@ static int tg3_mdio_init(struct tg3 *tp)
 	}
 
 	switch (phydev->drv->phy_id & phydev->drv->phy_id_mask) {
-	case TG3_PHY_ID_BCM57780:
+	case PHY_ID_BCM57780:
 		phydev->interface = PHY_INTERFACE_MODE_GMII;
 		phydev->dev_flags |= PHY_BRCM_AUTO_PWRDWN_ENABLE;
 		break;
-	case TG3_PHY_ID_BCM50610:
-	case TG3_PHY_ID_BCM50610M:
+	case PHY_ID_BCM50610:
+	case PHY_ID_BCM50610M:
 		phydev->dev_flags |= PHY_BRCM_CLEAR_RGMII_MODE |
 				     PHY_BRCM_RX_REFCLK_UNUSED |
 				     PHY_BRCM_DIS_TXCRXC_NOENRGY |
@@ -1152,11 +1152,11 @@ static int tg3_mdio_init(struct tg3 *tp)
 		if (tp->tg3_flags3 & TG3_FLG3_RGMII_EXT_IBND_TX_EN)
 			phydev->dev_flags |= PHY_BRCM_EXT_IBND_TX_ENABLE;
 		/* fallthru */
-	case TG3_PHY_ID_RTL8211C:
+	case PHY_ID_RTL8211C:
 		phydev->interface = PHY_INTERFACE_MODE_RGMII;
 		break;
-	case TG3_PHY_ID_RTL8201E:
-	case TG3_PHY_ID_BCMAC131:
+	case PHY_ID_RTL8201E:
+	case PHY_ID_BCMAC131:
 		phydev->interface = PHY_INTERFACE_MODE_MII;
 		phydev->dev_flags |= PHY_BRCM_AUTO_PWRDWN_ENABLE;
 		tp->tg3_flags3 |= TG3_FLG3_PHY_IS_FET;
@@ -1980,8 +1980,8 @@ static int tg3_phy_reset(struct tg3 *tp)
 		tg3_phy_toggle_apd(tp, false);
 
 out:
-	if ((tp->phy_id & TG3_PHY_ID_MASK) == TG3_PHY_ID_BCM50610 ||
-	    (tp->phy_id & TG3_PHY_ID_MASK) == TG3_PHY_ID_BCM50610M) {
+	if ((tp->phy_id & TG3_PHY_ID_MASK) == PHY_ID_BCM50610 ||
+	    (tp->phy_id & TG3_PHY_ID_MASK) == PHY_ID_BCM50610M) {
 		u32 reg;
 
 		/* Enable SM_DSP clock and tx 6dB coding. */
@@ -1994,8 +1994,8 @@ out:
 		tg3_phydsp_write(tp, MII_TG3_DSP_EXP8, reg);
 
 		/* Apply workaround to A0 revision parts only. */
-		if (tp->phy_id == TG3_PHY_ID_BCM50610 ||
-		    tp->phy_id == TG3_PHY_ID_BCM50610M) {
+		if (tp->phy_id == PHY_ID_BCM50610 ||
+		    tp->phy_id == PHY_ID_BCM50610M) {
 			tg3_phydsp_write(tp, 0x001F, 0x0300);
 			tg3_phydsp_write(tp, 0x601F, 0x0002);
 			tg3_phydsp_write(tp, 0x0F75, 0x003C);
@@ -2013,7 +2013,7 @@ out:
 		      MII_TG3_MISC_SHDW_RGMII_SEL;
 		tg3_writephy(tp, MII_TG3_MISC_SHDW, reg);
 	}
-	if ((tp->phy_id & TG3_PHY_ID_MASK) == TG3_PHY_ID_BCM57780) {
+	if ((tp->phy_id & TG3_PHY_ID_MASK) == PHY_ID_BCM57780) {
 		u32 reg;
 
 		/* Enable SM_DSP clock and tx 6dB coding. */
@@ -2651,11 +2651,11 @@ static int tg3_set_power_state(struct tg3 *tp, pci_power_t state)
 			phy_start_aneg(phydev);
 
 			phyid = phydev->drv->phy_id & phydev->drv->phy_id_mask;
-			if (phyid != TG3_PHY_ID_BCMAC131) {
-				phyid &= TG3_PHY_OUI_MASK;
-				if (phyid == TG3_PHY_OUI_1 ||
-				    phyid == TG3_PHY_OUI_2 ||
-				    phyid == TG3_PHY_OUI_3)
+			if (phyid != PHY_ID_BCMAC131) {
+				phyid &= PHY_BCM_OUI_MASK;
+				if (phyid == PHY_BCM_OUI_1 ||
+				    phyid == PHY_BCM_OUI_2 ||
+				    phyid == PHY_BCM_OUI_3)
 					do_low_power = true;
 			}
 		}
@@ -14485,11 +14485,11 @@ static char * __devinit tg3_phy_string(struct tg3 *tp)
 	case TG3_PHY_ID_BCM5718C:	return "5718C";
 	case TG3_PHY_ID_BCM5718S:	return "5718S";
 	case TG3_PHY_ID_BCM57765:	return "57765";
-	case TG3_PHY_ID_BCM50610:	return "50610";
-	case TG3_PHY_ID_BCM50610M:	return "50610M";
-	case TG3_PHY_ID_BCMAC131:	return "AC131";
-	case TG3_PHY_ID_BCM57780:	return "57780";
 	case TG3_PHY_ID_BCM8002:	return "8002/serdes";
+	case PHY_ID_BCM50610:		return "50610";
+	case PHY_ID_BCM50610M:		return "50610M";
+	case PHY_ID_BCMAC131:		return "AC131";
+	case PHY_ID_BCM57780:		return "57780";
 	case 0:			return "serdes";
 	default:		return "unknown";
 	}
diff --git a/drivers/net/tg3.h b/drivers/net/tg3.h
index 00e9347..62d898b 100644
--- a/drivers/net/tg3.h
+++ b/drivers/net/tg3.h
@@ -2959,15 +2959,12 @@ struct tg3 {
 #define TG3_PHY_ID_BCM5718S		0xbc050ff0
 #define TG3_PHY_ID_BCM57765		0x5c0d8a40
 #define TG3_PHY_ID_BCM5906		0xdc00ac40
-#define TG3_PHY_ID_BCM50610		0xbc050d60
-#define TG3_PHY_ID_BCM50610M		0xbc050d70
-#define TG3_PHY_ID_BCMAC131		0xbc050c70
-#define TG3_PHY_ID_RTL8211C		0x001cc910
-#define TG3_PHY_ID_RTL8201E		0x00008200
-#define TG3_PHY_ID_BCM57780		0x5c0d8990
 #define TG3_PHY_ID_BCM8002		0x60010140
 #define TG3_PHY_ID_INVALID		0xffffffff
 
+#define PHY_ID_RTL8211C			0x001cc910
+#define PHY_ID_RTL8201E			0x00008200
+
 #define TG3_PHY_ID_REV_MASK		0x0000000f
 #define TG3_PHY_REV_BCM5401_B0		0x1
 
@@ -2989,9 +2986,9 @@ struct tg3 {
 	 (X) == TG3_PHY_ID_BCM5755 || (X) == TG3_PHY_ID_BCM5756 || \
 	 (X) == TG3_PHY_ID_BCM5906 || (X) == TG3_PHY_ID_BCM5761 || \
 	 (X) == TG3_PHY_ID_BCM5718C || (X) == TG3_PHY_ID_BCM5718S || \
-	 (X) == TG3_PHY_ID_BCM57765 || (X) == TG3_PHY_ID_BCM57780 || \
-	 (X) == TG3_PHY_ID_BCM50610 || (X) == TG3_PHY_ID_BCM50610M || \
-	 (X) == TG3_PHY_ID_BCMAC131 || (X) == TG3_PHY_ID_BCM8002)
+	 (X) == TG3_PHY_ID_BCM57765 || (X) == PHY_ID_BCM57780 || \
+	 (X) == PHY_ID_BCM50610 || (X) == PHY_ID_BCM50610M || \
+	 (X) == PHY_ID_BCMAC131 || (X) == TG3_PHY_ID_BCM8002)
 
 	u32				led_ctrl;
 	u32				phy_otp;
diff --git a/drivers/net/tg3_compat.h b/drivers/net/tg3_compat.h
index 7bbe983..9307dfe 100644
--- a/drivers/net/tg3_compat.h
+++ b/drivers/net/tg3_compat.h
@@ -5,3 +5,35 @@
 #ifndef netdev_mc_empty
 #define netdev_mc_empty(dev) (netdev_mc_count(dev) == 0)
 #endif
+
+#ifndef PHY_ID_BCM50610
+#define PHY_ID_BCM50610		0xbc050d60
+#endif
+
+#ifndef PHY_ID_BCM50610M
+#define PHY_ID_BCM50610M	0xbc050d70
+#endif
+
+#ifndef PHY_ID_BCMAC131
+#define PHY_ID_BCMAC131		0xbc050c70
+#endif
+
+#ifndef PHY_ID_BCM57780
+#define PHY_ID_BCM57780		0x5c0d8990
+#endif
+
+#ifndef PHY_BCM_OUI_MASK
+#define PHY_BCM_OUI_MASK	0xfffffc00
+#endif
+
+#ifndef PHY_BCM_OUI_1
+#define PHY_BCM_OUI_1		0x00206000
+#endif
+
+#ifndef PHY_BCM_OUI_2
+#define PHY_BCM_OUI_2		0x0143bc00
+#endif
+
+#ifndef PHY_BCM_OUI_3
+#define PHY_BCM_OUI_3		0x03625c00
+#endif
-- 
1.7.3.4

